#!/usr/bin/env python3

# # I have neither given nor received any unauthorizedaid in completing this work,
# # nor have I presented someone else's work as my own.
# # Your Name: Devson Butani
# # LTU ID: 000732711
# # Date: 12/13/2022

PACKAGE = "term_project_pkg"  # Catkin Package Name

# Import library
from dynamic_reconfigure.parameter_generator_catkin import *

# Create Instance
gen = ParameterGenerator()

gen.add(
    "bw_threshold",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Black and White Threshold for mat detection",  # display on hover
    117,  # default
    0,  # min - all black
    255,  # max - all white
)
gen.add(
    "wheel_align_crop",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Aligns the dumping cups action forward/backward relative to the wheels",  # display on hover
    0.3,  # default
    0,  # min
    1,  # max
)
gen.add(
    "bot_crop",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Crop Down facing camera image by __'%' from the middle",  # display on hover
    0.5,  # default
    0,  # min
    1,  # max
)
gen.add(
    "bottom_crop",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Crop Down facing camera image by __'%' from the top",  # display on hover
    0.8,  # default
    0,  # min
    1,  # max
)
gen.add(
    "front_crop",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Crop Down facing camera image by __'%' from the front of the bot to use as plow area",  # display on hover
    0.45,  # default
    0,  # min
    1,  # max
)

### Debug display
gen.add(
    "edge_detect_debug",  # variable name
    bool_t,  # data type
    0,  # msg level
    ">>> Displays the node specific information",  # display on hover
    True,  # default
)

# Generate the menu in rqt tool
exit(gen.generate(PACKAGE, "inside_mat", "InsideMat"))
# 1st param - Package name constant from top
# 2nd param - node name this could be called in (for documentation only)
# 3rd param - This file's name without extension. Adds 'Config' at the end
#           - Use term_project_pkg.cfg and InsideMatConfig in py script

# for details: https://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
